compiler/rustc_middle/src/mir/type_foldable.rs

```diff
@@ -164,8 +170,11 @@ impl<'tcx> TypeFoldable<'tcx> for &'tcx ty::List<PlaceElem<'tcx>> {
         folder.tcx().intern_place_elems(&v)
     }
 
-    fn super_visit_with<V: TypeVisitor<'tcx>>(&self, visitor: &mut V) -> bool {
-        self.iter().any(|t| t.visit_with(visitor))
+    fn super_visit_with<V: TypeVisitor<'tcx>>(&self, visitor: &mut V) -> ControlFlow<(), ()> {
+        for t in self.iter() {
+            t.visit_with(visitor)?;
+        }
+        ControlFlow::CONTINUE
```

Comment:
```suggestion
        self.iter().map(|t| t.visit_with(visitor)).collect()
```

needs a `FromIterator` impl for `ControlFlow`, it's not automatic for `Try`, right?

Reply:
my idea would be to instead use `self.iter().try_for_each(|v| t.visit_with(visitor))`

Reply:
which imo is a bit clearer than implementing `FromIterator` here

Reply:
yea, I've seen this is actually done in other places in the PR. I did not know about `try_for_each`, and it does indeed seem more applicable here.